lib/api/src/modules/refs.ts

```diff
@@ -36,7 +36,7 @@ export interface ComposedRef {
   id: string;
   title?: string;
   url: string;
-  type?: 'auto-inject' | 'unknown' | 'lazy';
+  type?: 'auto-inject' | 'unknown' | 'lazy' | 'server-checked';
```

Comment:
How do the `auto-inject` and `lazy` types get set?

The "server-checkedness" feels sort of orthogonal to the type. But maybe i don't know what the type means.